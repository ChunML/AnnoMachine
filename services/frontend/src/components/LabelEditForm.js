import React from 'react';
import PropTypes from 'prop-types';

function LabelEditForm({ label, onInputChange }) {
  const inputStyle = {
    border: 'none',
    outline: 'none',
    borderBottom: '1px solid #38ACEC',
    borderRadius: '0',
    color: '#38ACEC',
    textAlign: 'center',
    fontSize: '1.2rem'
  };

  return (
    <div className='ui form'>
      <div className='field'>
        <input
          type='text'
          name='label'
          value={ label }
          onChange={ onInputChange }
          style={ inputStyle }
        />
      </div>
    </div>
  )
};

LabelEditForm.propTypes = {
  label: PropTypes.string.isRequired,
  onInputChange: PropTypes.func.isRequired
};

export default LabelEditForm;