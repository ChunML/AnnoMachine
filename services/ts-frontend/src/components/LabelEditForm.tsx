import React from 'react';
import PropTypes from 'prop-types';

interface LabelEditFormProps {
  label: string;
  onInputChange(e: React.FormEvent<HTMLInputElement>): void;
}

function LabelEditForm({ label, onInputChange }: LabelEditFormProps) {
  return (
    <div className="form">
      <div className="input-field text-center">
        <input
          type="text"
          name="label"
          value={label}
          onChange={onInputChange}
          // style={inputStyle}
        />
      </div>
    </div>
  );
}

LabelEditForm.propTypes = {
  label: PropTypes.string.isRequired,
  onInputChange: PropTypes.func.isRequired,
};

export default LabelEditForm;
