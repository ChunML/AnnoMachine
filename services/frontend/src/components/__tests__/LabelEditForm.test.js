import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import LabelEditForm from '../LabelEditForm';
import '../../setupTests';

const label = 'dog';

const onInputChange = jest.fn();

it('LabelEditForm renders properly', () => {
  const wrapper = shallow(
    <LabelEditForm label={label} onInputChange={onInputChange} />
  );
  const form = wrapper.find('.form');
  expect(form.length).toBe(1);
  const inputs = form.find('input');
  expect(inputs.length).toBe(1);
  expect(inputs.get(0).props.name).toEqual('label');
  expect(onInputChange).toHaveBeenCalledTimes(0);
  inputs.at(0).simulate('change', { target: { name: 'label', value: 'cat' } });
  expect(onInputChange).toHaveBeenCalledTimes(1);
});

it('LabelEditForm renders a snapshot properly', () => {
  const tree = renderer
    .create(<LabelEditForm label={label} onInputChange={jest.fn()} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
