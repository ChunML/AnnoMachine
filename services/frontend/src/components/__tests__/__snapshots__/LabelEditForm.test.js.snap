// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelEditForm renders a snapshot properly 1`] = `
<div
  className="ui form"
>
  <div
    className="field"
  >
    <input
      name="label"
      onChange={[MockFunction]}
      style={
        Object {
          "border": "none",
          "borderBottom": "1px solid #38ACEC",
          "borderRadius": "0",
          "color": "#38ACEC",
          "fontSize": "1.2rem",
          "outline": "none",
          "textAlign": "center",
        }
      }
      type="text"
      value="dog"
    />
  </div>
</div>
`;
